pkg/ruler/api.go

```diff
@@ -119,6 +125,26 @@ func respondError(logger log.Logger, w http.ResponseWriter, msg string) {
 	}
 }
 
+func respondClientError(logger log.Logger, w http.ResponseWriter, msg string) {
+	b, err := json.Marshal(&response{
+		Status:    "error",
+		ErrorType: v1.ErrServer,
+		Error:     msg,
+		Data:      nil,
+	})
+
+	if err != nil {
+		level.Error(logger).Log("msg", "error marshaling json response", "err", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
```

Comment:
I think the error should not be `http.StatusInternalServerError` here.

Reply:
[Prometheus](https://github.com/prometheus/prometheus/blob/main/web/api/v1/api.go#L1679) currently does this as well